### PR TITLE
Add additional tags to ItemMeta obfuscation filtering

### LINKED ISSUE #9451: [Feature] Better Offuscation
### Is your feature request related to a problem?

Currently obfuscation hides alot of stuff (attributes, lore, item-name, item-damage, enchantments) but I want it to hide more if possible.

### Describe the solution you'd like.

If possible, I want only CustomModelData to show and nothing else.

**People are able to use nbt inspectors** to see PublicBukkitValues and see if an item is unbreakable or not still.

### Other

current paper-world-defaults.yml

  obfuscation:
    items:
      hide-durability: true
      hide-itemmeta: true
      hide-itemmeta-with-visual-effects: false

new paper-world-defaults.yml?

  obfuscation:
    items:
      hide-durability: true
      hide-itemmeta: true
      hide-textures: false
      hide-itemmeta-with-visual-effects: false
      hide-other: true

### PR DESCRIPTION
**Pull request by Sytm**
Fixes #9451

(Is blacklist filtering even feasible like Machine said in the linked issue or would it be too much work to cover all edge cases?)

### CODE REVIEW

**Comment by electronicboy** [comment]
Filtering custom model data is a no go

**Review by Owen1212055 — DISMISSED** [review]
Removing CustomModelData will change the look of the item for other players. That should not be removed like that.

**Comment by Sytm** [comment]
That's why I deliberately put it inside the `hideItemmetaWithVisualEffects` block

**Comment by electronicboy** [comment]
visual effects are generally more trivial things, idk how a lode compass appears differently to others if it has a pos or not, but, CMD is purely intended to allow people to change how items look in general using datapacks, etc; it being removed ever sounds like a stupidly niche edge-case at best

File: patches/server/0732-Hide-unnecessary-itemmeta-from-clients.patch [hunk 1]
```diff
@@ -107,6 +111,8 @@ index a5cf00a41c61cc9f851781cc4f55c9bb47c2f5b6..f306c9e8f46d80d4c4fdc671e3754b3f
 +                // the position and fake the dimension
 +                tag.putString("LodestoneDimension", "paper:paper");
 +            }
++            // The id of a filled map
++            tag.remove("map");
```
Comment (NoahvdAa) [inline patches/server/0732-Hide-unnecessary-itemmeta-from-clients.patch]: What changes visually when this tag is missing?

**Review by NoahvdAa — COMMENTED** [review]
> idk how a lode compass appears differently to others if it has a pos or not

[it just shows as spinning](https://i.noah.pm/xWg6hF.gif)

File: patches/server/0732-Hide-unnecessary-itemmeta-from-clients.patch (hunk 1 above)
Comment (Sytm) [reply patches/server/0732-Hide-unnecessary-itemmeta-from-clients.patch]: Vanilla does not have any visual changes, but [some mods](https://modrinth.com/mod/not-enough-animations) which aren't necessarily cheats could behave differently.

**Review by Machine-Maker — APPROVED** [review]

**Review by lynxplay — APPROVED** [review]

**Pull request merged**